### PR TITLE
multimds: add commands to enable and set max_mds

### LINKED ISSUE #986: Add support for setting allow_multimds/max_mds
This would involve running these commands against the MDS after startup:

````
ceph fs set allow_multimds true --yes-i-really-mean-it
ceph fs set max_mds <n>
````

I think the tricky aspect of this is that it needs to happen after mds setup but requires the use of ceph admin keys (on the mons?). I'd be happy to implement this but I'm unsure how to do that.

Copied from: http://tracker.ceph.com/issues/17303

### PR DESCRIPTION
**Pull request by batrick**
Fixes #986.

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
Is this one really urgent? Because i'd like to have this one first, so you can adapt your code in consequence: https://github.com/ceph/ceph-ansible/pull/953

**Comment by batrick** [comment]
This can wait.

**Comment by leseb** [comment]
@batrick please rebase :)

**Comment by batrick** [comment]
@leseb done. I havne't had a chance to test it yet though.

File: roles/ceph-mon/tasks/create_mds_filesystems.yml [hunk 1]
```diff
@@ -15,3 +15,18 @@
   command: ceph --cluster {{ cluster }} fs new {{ cephfs }} {{ cephfs_metadata }} {{ cephfs_data }}
   changed_when: false
   when: ceph_release_num.{{ ceph_release }} > ceph_release_num.firefly
+
+- name: allow multimds
+  command: ceph --cluster {{ cluster }} fs set {{ cephfs }} allow_multimds true --yes-i-really-mean-it
+  changed_when: false
+  when:
+  - "{{ ceph_version | version_compare('10.1.2', '>=') }}"
```
Comment (leseb) [inline roles/ceph-mon/tasks/create_mds_filesystems.yml]: as per https://github.com/ceph/ceph-ansible/pull/953
this needs to be: `ceph_release_num.{{ ceph_release }} >= ceph_release_num.jewel`

File: roles/ceph-mon/tasks/create_mds_filesystems.yml [hunk 2]
```diff
@@ -15,3 +15,18 @@
...
+  - mds_allow_multimds
+
+- name: set max_mds
+  command: ceph --cluster {{ cluster }} fs set {{ cephfs }} max_mds {{ mds_max_mds }}
+  changed_when: false
+  when:
+  - "{{ ceph_version | version_compare('10.1.2', '>=') }}"
```
Comment (leseb) [inline roles/ceph-mon/tasks/create_mds_filesystems.yml]: ditto

File: roles/ceph-mon/tasks/create_mds_filesystems.yml (hunk 1 above)
Comment (leseb) [inline roles/ceph-mon/tasks/create_mds_filesystems.yml]: Arf, looks like a leftover, would you mind fixing this one too?

**Comment by batrick** [comment]
@leseb, done.

File: group_vars/all.sample [hunk 3]
```diff
@@ -260,6 +260,8 @@ dummy:
 ## MDS options
 #
 #mds_use_fqdn: false # if set to true, the MDS name used will be the fqdn in the ceph.conf
+#mds_allow_multimds: false
```
Comment (leseb) [inline group_vars/all.sample]: One last thing, please update roles/ceph-common/defaults/main.yml instead and run `generate_group_vars_sample.sh`, thanks

**Comment by batrick** [comment]
@leseb, okay done.

**Comment by leseb** [comment]
We are good now, thanks for your patience @batrick

**Pull request merged**